liquidity_pool/src/token_contract.rs

```diff
@@ -1,10 +1,10 @@
 use soroban_sdk::{Binary, Env, FixedBinary};
 use soroban_token_contract::public_types::U256;
 
-#[cfg(any(not(feature = "testutils"), feature = "token-wasm"))]
+#[cfg(any(not(any(test, feature = "testutils")), feature = "token-wasm"))]
 pub const TOKEN_CONTRACT: &[u8] = include_bytes!("../../soroban_token_contract.wasm");
 
-#[cfg(any(not(feature = "testutils"), feature = "token-wasm"))]
+#[cfg(any(not(any(test, feature = "testutils")), feature = "token-wasm"))]
```

Comment:
@jonjove This was pretty confusing to understand initially, so please let me know if I'm misunderstand how this is supposed to work.

I think the `testutils` cyclic dev-dependency hack was preventing any of the `token-wasm` feature usage to actually get used in tests, because tests always required testutils, and so there's no way `TOKEN_CONTRACT` could have been used in a test.

I'm not fixing that in this PR, I'm just filling in test gates for anything that was using testutils as a gate for whether tests were running or not. So I think a fix is needed if the hope is that `TOKEN_CONTRACT` will be used in tests.

Reply:
Ah, I think I see how this works now, and I'm updating this incorrectly, will update it.

Reply:
Ok, so no bug existed and I pushed a fix to this PR that I think updates the feature selection correctly. Still highly confusing.

Reply:
I think what you did is right. But I 100% agree that this is highly confusing. It was confusing when I wrote this, and it was confusing when I just reviewed these changes. Maybe there's a way to make this simpler. Maybe even just a comment would do the job.

Reply:
I'll add a comment as part of this PR.

Reply:
See b24f930.